budget_manager/views.py

```diff
@@ -61,4 +61,36 @@ class CategoryCreateView(LoginRequiredMixin, CreateView):
 
     def form_valid(self, form):
         form.instance.user = self.request.user
-        return super().form_valid(form)
\ No newline at end of file
+        return super().form_valid(form)
+
+
+
+class IncomeCreateView(LoginRequiredMixin, CreateView):
+    model = Income
+    template_name = 'income_form.html'
+    fields = ['amount', 'source', 'income_date', 'currency']
+    success_url = reverse_lazy('income-list')
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        return super().form_valid(form)
+
+
+class IncomeListView(LoginRequiredMixin, ListView):
+    model = Income
+    template_name = 'income_list.html'
+    fields = ['amount', 'source', 'income_date', 'currency']
+
+
+class IncomeUpdateView(LoginRequiredMixin, UpdateView):
+    model = Income
+    template_name = 'income_update.html'
+    fields = ['amount', 'source', 'income_date', 'currency']
+    success_url = reverse_lazy('income-list')
+    context_object_name = 'income'
```

Comment:
jeśli tutaj dodajesz context_object_name to dawałbym już wszędzie, wtedy będzie wiadomo, gdzie taką nazwę szukać